Assets/MixedRealityToolkit/Utilities/AssetScriptReferenceRetargeter.cs

```diff
@@ -0,0 +1,593 @@
+                }
+            }
+
+            foreach (var extension in foundNonYamlExtensions)
+            {
+                Debug.Log($"Not a YAML extension: {extension}");
+            }
+
+            var tasks = yamlAssets.Select(t => Task.Run(() => ProcessYamlFile(t.Item1, t.Item2, remapDictionary)));
+            Task.WhenAll(tasks).Wait();
+
+            PostProcess(outputDirectory, dllGuids);
+        }
+
+        private static async Task ProcessYamlFile(string filePath, string targetPath, Dictionary<string, Tuple<string, long>> remapDictionary)
+        {
+            int lineNum = 0;
+            using (StreamReader reader = new StreamReader(filePath))
+            using (StreamWriter writer = new StreamWriter(targetPath))
+            {
+                while (!reader.EndOfStream)
+                {
+                    string line = await reader.ReadLineAsync();
+                    lineNum++;
+                    if (line.Contains("m_Script"))
+                    {
+                        if (!line.Contains('}'))
+                        {
+                            // Read the second line as well
+                            line += await reader.ReadLineAsync();
+                            lineNum++;
+
+                            if (!line.Contains('}'))
+                            {
+                                throw new InvalidDataException($"Unexpected part of YAML line split over more than two lines, starting two lines: {line}");
+                            }
+                        }
+
+                        if (line.Contains(ScriptFileIdConstant))
+                        {
+                            Match regexResults = Regex.Match(line, @"guid:\s*([0-9a-fA-F]*)");
+                            if (!regexResults.Success || regexResults.Groups.Count != 2 || !regexResults.Groups[1].Success || regexResults.Groups[1].Captures.Count != 1)
+                            {
+                                throw new InvalidDataException($"Failed to find the guid in line: {line}.");
+                            }
+
+                            string guid = regexResults.Groups[1].Captures[0].Value;
+                            if (remapDictionary.TryGetValue(guid, out Tuple<string, long> tuple))
+                            {
+                                line = $"  m_Script: {{fileID: {tuple.Item2}, guid: {tuple.Item1}, type: 3}}";
+                            }
+                            else if (nonClassDictionary.ContainsKey(guid))
+                            {
+                                throw new InvalidDataException($"A script without a class ({nonClassDictionary[guid]}) is being processed.");
+                            }
+                            else
+                            {
+                                // Switch to error later
+                                Debug.LogWarning($"Couldn't find a script remap for {guid} in file: '{filePath}' at line '{lineNum}'.");
```

Comment:
>LogWarning [](start = 38, length = 10)

Nit: Maybe LogError and delete the comment on the line above.

Reply:
Not yet, this will give lot's of false negatives still.

---
In reply to: [277834376](https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/4051#discussion_r277834376) [](ancestors = 277834376)